identity_iota_core/src/state_metadata/document.rs

```diff
@@ -93,26 +95,47 @@ impl StateMetadataDocument {
       identity_did::Error::InvalidDocument("expected data to have at least length 5", None),
     )?)?;
 
-    let inner: &[u8] = data.get(5..).ok_or(identity_did::Error::InvalidDocument(
-      "expected data to have at least length 6",
-      None,
-    ))?;
+    let data_len_packed: [u8; 2] = data
+      .get(5..=6)
+      .ok_or(identity_did::Error::InvalidDocument(
+        "expected data to have at least length 8",
+        None,
+      ))?
```

Comment:
I think I wrote those, but we can probably improve the error messages. I think we should include what we're doing (`state metatdata decoding`) because we're reusing the fairly generic error from `identity_did`, or introduce a separate, more specific `StateMetadataDecodingError` or something. Perhaps using a style along the lines of:

```
state metatdata decoding: expected (DID marker/encoding/version/...) at offset [x..=y]
```

E.g.

```suggestion
    let data_len_packed: [u8; 2] = data
      .get(5..=6)
      .ok_or(identity_did::Error::InvalidDocument(
        "state metadata decoding: expected data length at offset [5..=6]",
        None,
      ))?
```